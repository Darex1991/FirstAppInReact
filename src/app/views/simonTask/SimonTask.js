import React, {Component} from 'react';
import cx             from 'classnames';
import shallowCompare from 'react-addons-shallow-compare';
import {Form, Text} from 'react-form';
import ProfileComponent from './ProfileComponent';
import NotificationComponent from './NotificationComponent';
import ColorComponent from './ColorComponent';
import PhoneNumbersComponent from './PhoneNumbersComponent';

export default class SimonTask extends Component {
  constructor(props) {
    super(props);
    this.state = {
      animated: true,
      viewEntersAnim: true,
      dataBase: {
        phones: [{id:1, name: 'asd', key: 1, value: 'val 1'},{id:2, name: 'asdasdasd', key: 2, value: 'val 122222'},{id:3, name: 'asdasdasd', key: 3, value: 'val asdasd'},{id:4, name: '222', key: 4, value: 'val213123 1'}]
      }
    };
  }

  shouldComponentUpdate = (nextProps, nextState) => {
    return shallowCompare(this, nextProps, nextState);
  };

  render() {
    const {animated, viewEntersAnim} = this.state;
    return (
      <div
        className={cx({
          'animatedViews': animated,
          'view-enter': viewEntersAnim
        })}>
        <h1>
          SimonTask
        </h1>
        <Form
          onSubmit={(values) => {
            console.log('Success!', values)
          }}
          validate={ values => {
            const { firstName, lastName, email, password, password2} = values;
            return {
                firstName: !firstName ? 'First name is required' : undefined,
                lastName: !lastName ? 'Last name is required' : undefined,
                email: !email ? 'Email name is required' : undefined,
                password: (() => {
                  if (!password || password != password2) {
                    if (!password){ return 'Password confirm name is required' }
                    else { return `Password confirm must be the same` }
                  }
                })(),
                password2: (() => {
                  if (!password2 || password != password2) {
                    if (!password2){ return 'Password confirm name is required' }
                    else { return `Password confirm must be the same` }
                  }
                })()
              }
          }}
          defaultValues={{
            emailCheckboxPremium: true,
            textCheckboxPremium: true,
            text2CheckboxPremium: true
          }}
        >
          {({submitForm}) => {
            return (
              <form onSubmit={submitForm}>
                <ProfileComponent />
                <PhoneNumbersComponent dataBase={this.state.dataBase} />
                <NotificationComponent dataBase={this.state.dataBase} />
                <ColorComponent />
                <button type='submit'>Submit</button>
              </form>
            )
          }}

        </Form>
      </div>
    );
  }
};
