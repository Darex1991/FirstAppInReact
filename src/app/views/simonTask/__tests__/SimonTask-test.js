import React from 'react';
import renderer from 'react-test-renderer';
import SimonTask from '../SimonTask';
test('Should render SimonTask correctly', () => {
    const tree = renderer.create(
        <SimonTask /> 
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
    