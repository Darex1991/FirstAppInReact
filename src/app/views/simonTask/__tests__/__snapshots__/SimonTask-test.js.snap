exports[`test Should render SimonTask correctly 1`] = `
<div
  className="animatedViews view-enter">
  <h1>
     SimonTask 
  </h1>
  <div
    className="ReactForm">
    <form
      onSubmit={[Function]}>
      <div
        className="profile default-input">
        <h2>
           Profile 
        </h2>
        <div
          className="row">
          <div
            className="col-xs-6 m--b">
            <label>
               FirstName 
            </label>
            <div
              className="FormInput">
              <input
                onBlur={[Function]}
                onChange={[Function]}
                value="" />
              <div
                className="FormError"
                style={
                  Object {
                    "display": "none",
                  }
                }>
                
              </div>
            </div>
          </div>
          <div
            className="col-xs-6 m--b">
            <label>
               LastName 
            </label>
            <div
              className="FormInput">
              <input
                onBlur={[Function]}
                onChange={[Function]}
                value="" />
              <div
                className="FormError"
                style={
                  Object {
                    "display": "none",
                  }
                }>
                
              </div>
            </div>
          </div>
          <div
            className="col-xs-12 m--b">
            <label>
               Email 
            </label>
            <div
              className="FormInput">
              <input
                onBlur={[Function]}
                onChange={[Function]}
                type="email"
                value="" />
              <div
                className="FormError"
                style={
                  Object {
                    "display": "none",
                  }
                }>
                
              </div>
            </div>
          </div>
          <div
            className="col-xs-6 m--b">
            <div
              className="FormInput">
              <input
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value="" />
              <div
                className="FormError"
                style={
                  Object {
                    "display": "none",
                  }
                }>
                
              </div>
            </div>
          </div>
          <div
            className="col-xs-6 m--b">
            <div
              className="FormInput">
              <input
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Password confirm"
                type="password"
                value="" />
              <div
                className="FormError"
                style={
                  Object {
                    "display": "none",
                  }
                }>
                
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="phone default-input">
        <h2>
           Phone 
        </h2>
        <div>
          <button
            onClick={[Function]}>
            Add new
          </button>
        </div>
        <div
          className="col-xs-12">
          <div
            className="col-xs-3">
            <button
              disabled={true}
              onClick={[Function]}>
               GO UP 
            </button>
            <button
              disabled={false}
              onClick={[Function]}>
               GO DOWN 
            </button>
          </div>
          <div
            className="col-xs-3">
            asd
          </div>
          <div
            className="col-xs-3">
            <div
              className="FormInput -hasError">
              <input
                onBlur={[Function]}
                onChange={[Function]}
                value="val 1" />
              <div
                className="FormError"
                style={
                  Object {
                    "display": "block",
                  }
                }>
                
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-xs-12">
          <div
            className="col-xs-3">
            <button
              disabled={false}
              onClick={[Function]}>
               GO UP 
            </button>
            <button
              disabled={false}
              onClick={[Function]}>
               GO DOWN 
            </button>
          </div>
          <div
            className="col-xs-3">
            asdasdasd
          </div>
          <div
            className="col-xs-3">
            <div
              className="FormInput -hasError">
              <input
                onBlur={[Function]}
                onChange={[Function]}
                value="val 122222" />
              <div
                className="FormError"
                style={
                  Object {
                    "display": "block",
                  }
                }>
                
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-xs-12">
          <div
            className="col-xs-3">
            <button
              disabled={false}
              onClick={[Function]}>
               GO UP 
            </button>
            <button
              disabled={false}
              onClick={[Function]}>
               GO DOWN 
            </button>
          </div>
          <div
            className="col-xs-3">
            asdasdasd
          </div>
          <div
            className="col-xs-3">
            <div
              className="FormInput -hasError">
              <input
                onBlur={[Function]}
                onChange={[Function]}
                value="val asdasd" />
              <div
                className="FormError"
                style={
                  Object {
                    "display": "block",
                  }
                }>
                
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-xs-12">
          <div
            className="col-xs-3">
            <button
              disabled={false}
              onClick={[Function]}>
               GO UP 
            </button>
            <button
              disabled={true}
              onClick={[Function]}>
               GO DOWN 
            </button>
          </div>
          <div
            className="col-xs-3">
            222
          </div>
          <div
            className="col-xs-3">
            <div
              className="FormInput -hasError">
              <input
                onBlur={[Function]}
                onChange={[Function]}
                value="val213123 1" />
              <div
                className="FormError"
                style={
                  Object {
                    "display": "block",
                  }
                }>
                
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="profile default-input">
        <h2>
          Notification
        </h2>
        <div>
          <table>
            <tbody>
              <tr>
                <th>
                   
                </th>
                <th>
                  Standard Prior
                </th>
                <th>
                  High Prior
                </th>
              </tr>
              <tr>
                <td>
                  <p>
                     Email 
                    <span>
                      (asdasd@asdasd.pl)
                    </span>
                  </p>
                </td>
                <td>
                  <div
                    className="FormInput">
                    <input
                      checked={false}
                      onBlur={[Function]}
                      onChange={[Function]}
                      type="checkbox" />
                    <div
                      className="FormError"
                      style={
                        Object {
                          "display": "none",
                        }
                      }>
                      
                    </div>
                  </div>
                </td>
                <td>
                  <div
                    className="FormInput">
                    <input
                      checked={true}
                      onBlur={[Function]}
                      onChange={[Function]}
                      type="checkbox" />
                    <div
                      className="FormError"
                      style={
                        Object {
                          "display": "none",
                        }
                      }>
                      
                    </div>
                  </div>
                </td>
              </tr>
              <tr>
                <td>
                  <p>
                    asd
                     
                    <span>
                      (
                      val 1
                      ) 
                    </span>
                  </p>
                </td>
                <td>
                  <input
                    checked={true}
                    onChange={[Function]}
                    type="checkbox" />
                </td>
                <td>
                  <input
                    checked={true}
                    onChange={[Function]}
                    type="checkbox" />
                </td>
              </tr>
              <tr>
                <td>
                  <p>
                    asdasdasd
                     
                    <span>
                      (
                      val 122222
                      ) 
                    </span>
                  </p>
                </td>
                <td>
                  <input
                    checked={false}
                    onChange={[Function]}
                    type="checkbox" />
                </td>
                <td>
                  <input
                    checked={false}
                    onChange={[Function]}
                    type="checkbox" />
                </td>
              </tr>
              <tr>
                <td>
                  <p>
                    asdasdasd
                     
                    <span>
                      (
                      val asdasd
                      ) 
                    </span>
                  </p>
                </td>
                <td>
                  <input
                    checked={false}
                    onChange={[Function]}
                    type="checkbox" />
                </td>
                <td>
                  <input
                    checked={true}
                    onChange={[Function]}
                    type="checkbox" />
                </td>
              </tr>
              <tr>
                <td>
                  <p>
                    222
                     
                    <span>
                      (
                      val213123 1
                      ) 
                    </span>
                  </p>
                </td>
                <td>
                  <input
                    checked={true}
                    onChange={[Function]}
                    type="checkbox" />
                </td>
                <td>
                  <input
                    checked={false}
                    onChange={[Function]}
                    type="checkbox" />
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        className="profile default-input">
        <h2>
          Color Schema
        </h2>
        <div>
          <div
            className="FormInput">
            <input
              checked={false}
              onBlur={[Function]}
              onChange={[Function]}
              type="checkbox" />
            <div
              className="FormError"
              style={
                Object {
                  "display": "none",
                }
              }>
              
            </div>
          </div>
          <p>
            Dark schema
          </p>
        </div>
      </div>
      <button
        type="submit">
         Submit 
      </button>
    </form>
  </div>
</div>
`;
